src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,162 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+        first = new Node<>(null, null, null);
+        last = new Node<>(null, null, null);
+    }
+
+    private void checkIndexException(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException();
+        }
+    }
+
+    private void addToTail(T value) {
+        Node<T> newElem = new Node<>(last, value, null);
+        if (last.item == null) {
+            first.next = newElem;
+        } else {
+            last.next = newElem;
+        }
+        last = newElem;
+        size++;
+    }
+
+    private void addToHead(T value) {
+        Node<T> newElem = new Node<>(null, value, first);
+        if (first.item == null) {
+            last.prev = newElem;
+        } else {
+            first.prev = newElem;
+        }
+        first = newElem;
+        size++;
+    }
+
     @Override
     public void add(T value) {
-
+        if (first.item == null) {
+            addToHead(value);
+        } else {
+            addToTail(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            checkIndexException(index);
```

Comment:
@Korfat should you use checkIndexException inside if statement or instead of it?